web_tool_script_2.R

```diff
@@ -45,161 +45,185 @@ analysis_inputs_path <- set_analysis_inputs_path(twodii_internal, data_location_
 # delete all results files
 unlink(paste0(results_path,"/*"), force = TRUE, recursive = TRUE)
 
+# run again so output folders are available after deleting past results
 file_names <- read_csv(paste0(proc_input_path, "/file_names.csv"))
+create_portfolio_subfolders(file_names)
 
 port_col_types <- set_col_types(grouping_variables, "ddddccccddcl")
 
 ##################
 ##### EQUITY #####
 ##################
 
-equity_input_file <- paste0(proc_input_path, "/",file_names$loc_name[1],"/",project_name, "_equity_portfolio.rda")
-
-if(file.exists(equity_input_file)){
-  port_raw_all_eq <- read_rds(equity_input_file) %>% 
-    mutate(id = as.character(id))
-  
-  if(length(colnames(port_raw_all_eq)) != nchar(port_col_types)){stop("Check port_col_types: difference in length")}
-  
-  ald_scen_eq <- get_ald_scen("Equity")
+for (p in 1:nrow(file_names)){
   
-  ald_raw_eq <- get_ald_raw("Equity")
+  equity_input_file <- paste0(proc_input_path, "/",file_names$portfolio_name[p],"/equity_portfolio.rda")
+  portfolio_name_ <- file_names$portfolio_name[p]
   
-  list_investors_eq <- unique(port_raw_all_eq$investor_name)
-  
-  for (e in 1:length(list_investors_eq) ){
-    map_eq <- NA
-    company_all_eq <- NA
-    port_all_eq <- NA
-    
-    investor_name_select <- list_investors_eq[e]
-    print(paste0(e, ": ", investor_name_select))
-    
-    port_raw_eq <- port_raw_all_eq %>% filter(investor_name == investor_name_select)
-    
-    port_eq <- calculate_weights(port_raw_eq, "Equity", grouping_variables)
-    
-    port_eq <- merge_in_ald(port_eq, ald_scen_eq)
-    
-    # Portfolio weight methodology
-    port_pw_eq <- port_weight_allocation(port_eq)
-    
-    company_pw_eq <- aggregate_company(port_pw_eq)
+  if(file.exists(equity_input_file)){
+    port_raw_all_eq <- read_rds(equity_input_file) %>% 
+      mutate(id = as.character(id))
     
-    port_pw_eq <- aggregate_portfolio(company_pw_eq) 
+    if(length(colnames(port_raw_all_eq)) != nchar(port_col_types)){stop("Check port_col_types: difference in length")}
     
-    # Ownership weight methodology
-    port_own_eq <- ownership_allocation(port_eq)
+    ald_scen_eq <- get_ald_scen("Equity")
     
-    company_own_eq <- aggregate_company(port_own_eq)
+    ald_raw_eq <- get_ald_raw("Equity")
     
-    port_own_eq <- aggregate_portfolio(company_own_eq) 
+    list_investors_eq <- unique(port_raw_all_eq$investor_name)
     
-    # Create combined outputs
-    company_all_eq <- bind_rows(company_pw_eq, company_own_eq)
-    
-    port_all_eq <- bind_rows(port_pw_eq, port_own_eq) 
-    
-    if(has_map){
+    for (e in 1:length(list_investors_eq) ){
```

Comment:
Should this not be "portfolio_name"? and list_portfolios_eq ?

Reply:
I don't think that's necessary. I wrapped a whole other loop around the calculations that goes through the file names, which are, at east in this case, the portfolios. I left this ne in here, just in case there actually is more than one investor in a file. But I think it could just as well be removed